app/models/service_template_ansible_playbook.rb

```diff
@@ -24,13 +24,41 @@ def self.default_retirement_entry_point
   #     :reconfigure (same as provision)
   #
   def self.create_catalog_item(options, _auth_user)
+    task_id = create_catalog_item_queue(options, _auth_user)
+    task = MiqTask.wait_for_taskid(task_id)
+    task.task_results
```

Comment:
@syncrou 
The wait_for_taskid can fail when we catch any exceptions on the worker side and propagate it to the caller via the task.status. so we should be checking for errors.
For specs you could mimic this behavior by passing an invalid playbook_id and the worker side should be raising an exception and the task.status will get set appropriately.